napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1521 @@
+        elif mode == 'add_path':
+            self.cursor = 'cross'
+            self.interactive = False
+            self.help = ('hold <space> to pan/zoom, '
+                         'press <esc> to finish drawing')
+            self.status = mode
+            self._mode = mode
+        elif mode == 'add_polygon':
+            self.cursor = 'cross'
+            self.interactive = False
+            self.help = ('hold <space> to pan/zoom, '
+                         'press <esc> to finish drawing')
+            self.status = mode
+            self._mode = mode
+        elif mode == 'vertex_insert':
+            self.cursor = 'cross'
+            self.interactive = False
+            self.help = 'hold <space> to pan/zoom'
+            self.status = mode
+            self._mode = mode
+        elif mode == 'vertex_remove':
+            self.cursor = 'cross'
+            self.interactive = False
+            self.help = 'hold <space> to pan/zoom'
+            self.status = mode
+            self._mode = mode
+        else:
+            raise ValueError("Mode not recongnized")
+
+        self.events.mode(mode=mode)
+        if not ((mode == 'vertex_insert' or mode == 'vertex_remove' or
+                mode == 'direct' or mode == 'select') and
+                (old_mode == 'vertex_insert' or old_mode == 'vertex_remove' or
+                old_mode == 'direct' or old_mode == 'select')):
+            self._finish_drawing()
+        self.refresh()
+
+    def _get_shape(self):
+        """Determines the shape of the vertex data.
+        """
+        if len(self.data._vertices) == 0:
+            return [1, 1]
+        else:
+            return np.max(self.data._vertices, axis=0) + 1
+
+    def _update(self):
+        """Update the underlying visual.
+        """
+        if self._need_display_update:
+            self._need_display_update = False
+            self._set_view_slice()
+
+        if self._need_visual_update:
+            self._need_visual_update = False
+            self._node.update()
+
+    def _refresh(self):
+        """Fully refresh the underlying visual.
+        """
+        self._need_display_update = True
+        self._update()
+
+    def _set_view_slice(self):
+        """Sets the shape mesh data to the view.
+        """
+        z_order = self.data._mesh_triangles_z_order
+        faces = self.data._mesh_triangles[z_order]
+        colors = self.data._mesh_triangles_colors[z_order]
+        vertices = self.data._mesh_vertices
+        if len(faces) == 0:
+            self._node._subvisuals[3].set_data(vertices=None, faces=None)
+        else:
+            self._node._subvisuals[3].set_data(vertices=vertices, faces=faces,
+                                               face_colors=colors)
+        self._need_visual_update = True
+        self._set_highlight(force=True)
+        self._update()
+
+    def bounding_box(self, index):
+        """Creates the bounding box around a shape or list of shapes. If a
```

Comment:
```suggestion
        """Create the bounding box around a shape or list of shapes.
```

Also, after, two newlines after the period and then `If a` starts a new paragraph.

Reply:
tried to fix, was a little unclear what you meant with `If a`